File: test-suite-kotlin/src/test/kotlin/io/micronaut/docs/aws/alexa/flashbriefing/FlashBriefingsController.kt
package io.micronaut.docs.aws.alexa.flashbriefing

//tag::clazz[]
import io.micronaut.aws.alexa.flashbriefing.FlashBriefingItem
import io.micronaut.http.annotation.Controller
import io.micronaut.http.annotation.Get
import jakarta.validation.Validator

@Controller("/news")
class FlashBriefingsController(
    private val validator: Validator,
    private val flashBriefingRepository: FlashBriefingRepository
) {

    @Get // <1>
    fun index(): List<FlashBriefingItem> {
        return flashBriefingRepository.find()
            .filter { item -> validator.validate(item).isEmpty() } // <2>
            .sorted() // <3>
            .take(5) // <4>
    }

}
//end::clazz[]

File: test-suite-python/src/test/python/micronaut/docs/aws/alexa/flashbriefing/FlashBriefingsController.py
# tag::clazz[]
from functools import cmp_to_key

from jakarta.validation import Validator
from micronaut.aws.alexa.flashbriefing import FlashBriefingItem
from micronaut.http.annotation import Controller, Get

from .FlashBriefingRepository import FlashBriefingRepository


@Controller("/news")
class FlashBriefingsController:

    def __init__(self, validator: Validator, flash_briefing_repository: FlashBriefingRepository):
        self.validator = validator
        self.flash_briefing_repository = flash_briefing_repository

    @Get  # <1>
    def index(self) -> list[FlashBriefingItem]:
        items = [item for item in self.flash_briefing_repository.find()
                 if self.validator.validate(item).isEmpty()]  # <2>
        items.sort(key=cmp_to_key(lambda a, b: a.compareTo(b)))  # <3>
        return items[:5]  # <4>
# end::clazz[]
